tests/CMakeLists.txt

```diff
@@ -43,24 +43,30 @@ if (GTest_FOUND)
   include_directories(SYSTEM ${GTEST_INCLUDE_DIRS})
 else ()
   message(STATUS "GTest not found on the system: will use GTest bundled with this source")
-  add_subdirectory(gtest)
-  include_directories(SYSTEM "${gtest_SOURCE_DIR}/include" "${gtest_SOURCE_DIR}")
 
-  # Emulate the FindGTest module's variable.
-  set(GTEST_LIBRARIES gtest)
-
-  # Ignore some warnings when building gtest binaries.
-  if(NOT MSVC)
-    set_property(TARGET gtest
-      APPEND_STRING
-      PROPERTY
-        COMPILE_FLAGS " -Wno-undef -Wno-sign-compare")
```

Comment:
These seem to be worth keeping maybe. Possibly in the inner CMakeLists.txt if the change to ExternalProject makes it impossible to set properties here.